src/Logic.hs

```diff
@@ -354,26 +356,32 @@ handleCommentAdded
   -> Text
   -> ProjectState
   -> Action ProjectState
-handleCommentAdded triggerConfig pr author body state =
-  if Pr.existsPullRequest pr state
+handleCommentAdded triggerConfig prId author body state =
+  let maybePR = Pr.lookupPullRequest prId state in
+  case maybePR of
     -- Check if the commment is a merge command, and if it is, check if the
     -- author is allowed to approve. Comments by users with push access happen
     -- frequently, but most comments are not merge commands, and checking that
     -- a user has push access requires an API call.
-    then do
+    Just pr -> do
       let approvalType = parseMergeCommand triggerConfig body
       isApproved <- if isJust approvalType
         then isReviewer author
         else pure False
       if isApproved
         -- The PR has now been approved by the author of the comment.
-        then
+        then do
           let (order, state') = Pr.newApprovalOrder state
-          in approvePullRequest pr (Approval author (fromJust approvalType) order) state'
+          state'' <- approvePullRequest prId (Approval author (fromJust approvalType) order) state'
+          -- Check whether the integration branch is master, if not, mark the integration as forbidden
+          -- NOTE: Maybe the integration branch should be configurable per project?
+          if Pr.baseBranch pr /= Branch "master" 
```

Comment:
The `master` branch is hardcoded here, but it is already part of the config. See `ProjectConfiguration.branch`

Reply:
I think I've resolved this.